### PR TITLE
implement search function and http response with json function

### PR DESCRIPTION
**Pull request by Hang91**

### CODE REVIEW

File: dao.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	"gopkg.in/mgo.v2/bson"
+)
+
+// Users data access object
+type UsersDAO struct {
+	Server   string
+	Database string
+}
+
+var db *mgo.Database
```
Comment (Syulei) [inline dao.go:16]: mgo.Database is an address?

File: dao.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	if err != nil {
+		log.Println(err)
+		return
+	}
+	log.Println("Database", u.Database, "connected.")
+	db = session.DB(u.Database)
+}
+
+// Searchword
+func (u *UsersDAO) SearchWord(spell string) (bool, Word) {
```
Comment (Syulei) [inline dao.go:39]: func (u *UsersDAO) SearchWord (spell String)  (bool Word)

File: main.go [hunk 3]
```diff
@@ -4,36 +4,42 @@ import (
...
-	"../word-memory/user"
-
 	"github.com/gorilla/mux"
 )
 
+var dao UsersDAO
+
 func main() {
 	router := mux.NewRouter()
+	dao = UsersDAO{
```
Comment (Syulei) [inline main.go:14]: dao := UsersDAO?

File: respond.go [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+func respondWithError(w http.ResponseWriter, code int, msg string) {
+	respondWithJson(w, code, map[string]string{"error": msg})
```
Comment (Syulei) [inline respond.go:9]: string{"error": msg} ?

File: user.go [hunk 5]
```diff
@@ -1,12 +1,23 @@
...
 	"fmt"
+	"log"
 	"net/http"
+
+	"github.com/gorilla/mux"
 )
 
 func SearchWord(w http.ResponseWriter, r *http.Request) {
-	fmt.Fprintln(w, "not implemented yet!")
+	params := mux.Vars(r)
```
Comment (Syulei) [inline user.go:12]: params works as a router to connect

File: user.go [hunk 6]
```diff
@@ -1,12 +1,23 @@
...
+	flag, word := dao.SearchWord(params["spell"])
+	if !flag {
+		log.Println("word not found")
+		respondWithError(w, http.StatusBadRequest, "No such words")
+		return
+	}
+	respondWithJson(w, http.StatusOK, word.Meaning)
 }
 
 func UserLogIn(w http.ResponseWriter, r *http.Request) {
```
Comment (Syulei) [inline user.go:23]: function name format

Comment (Hang91) [reply user.go:23]: I will change the first letter of function name to uppercase

**Pull request merged**